### PR TITLE
Add SchemaManager test

### PR DESCRIPTION
**Pull request by kitagry**

### CODE REVIEW

File: bqemulatormanager/schema.py [hunk 1]
```diff
@@ -1,48 +1,69 @@
...
+
+    params:
+        schema_file_path:
+            path to yaml file which contains schema of tables in emulator.
+
+        client:
+            client of production BigQuery. If you set this, SchemaManager will fetch schema from production BigQuery.
+    """
+
+    def __init__(self, schema_file_path: Optional[str] = "master_schema.yaml", client: Optional[BigQueryClient] = None):
```
Comment (hirosassa) [inline bqemulatormanager/schema.py]: [IMO, nits] Current default schema file name is too common to hit the other kind of file unintentionally.  How about change it to `bqem_master_schema.yaml`?

**Review by hirosassa — APPROVED** [review]
Looks good except for a minor comment! Thanks!

File: bqemulatormanager/schema.py [hunk 2]
```diff
@@ -1,48 +1,69 @@
 import os
-from typing import List
```
Comment (yokomotod) [inline bqemulatormanager/schema.py:2]: [imo]

I like `list` , `str | None` with `from __future__ import annotations`

File: test/test_schema.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    def get_table(self, _table: str) -> Table:  # noqa: U101
+        return self.__table
+
+
+class TestSchemaManager(unittest.TestCase):
+    test_schema_dir = ".testing"
+
+    def setUp(self) -> None:
+        os.mkdir(self.test_schema_dir)
```
Comment (yokomotod) [inline test/test_schema.py]: [fyi] you can use python `tempfile` as well

File: bqemulatormanager/schema.py [hunk 4]
```diff
@@ -1,48 +1,69 @@
...
+    def get_schema(self, project_id: str, dataset_id: str, table_id: str) -> List[bigquery.SchemaField]:
+        schema = self.__master_schema.get(project_id, {}).get(dataset_id, {}).get(table_id, {})
 
         if schema == {}:
-            schema = self._get_schema_from_production(table_id)
-            deepupdate(self.master_schema, {project: {dataset: {table: [s._properties for s in schema]}}})
-            self.change_flg = True
+            schema = self._get_schema_from_production(project_id, dataset_id, table_id)
+            if schema == []:
+                raise SchemaError(f"schema for {project_id}.{dataset_id}.{table_id} is not found in master schema and production environment")
```
Comment (yokomotod) [inline bqemulatormanager/schema.py]: [nits][imo]

`_get_schema_from_production()` should raise this error, rather than return `[]` ?

**Review by yokomotod — APPROVED** [review]
@kitagry lgtm with some comments

File: bqemulatormanager/schema.py [hunk 5]
```diff
@@ -1,48 +1,75 @@
+from __future__ import annotations
+
 import os
-from typing import List
```
Comment (yokomotod) [inline bqemulatormanager/schema.py:2]: @kitagry [fyi] neither `Dict` nor `List` are required to import, you can use `list` `dict` (linter....)

Comment (kitagry) [reply bqemulatormanager/schema.py:2]: @yokomotod  Python3.8 cannot define the type.

> TypeError: 'type' object is not subscriptable

https://github.com/m3dev/bqemulatormanager/actions/runs/5286103390/jobs/9565213830

So, we need to import List and DIct

Comment (kitagry) [reply bqemulatormanager/schema.py:2]: I defined 3 types.

```python
TableSchema = List[Dict[str, Any]]
DatasetSchema = Dict[str, TableSchema]
ProjectSchema = Dict[str, DatasetSchema]
```

**Pull request merged**